### PR TITLE
fix(@angular-devkit/build-angular): never use component css sourcemap…

### PR DESCRIPTION
**Pull request by filipesilva**
… when optimizations are on.

It will just increase bundle size without offering good debug experience.

### CODE REVIEW

**Comment by filipesilva** [comment]
According to @IgorMinar:

> If you are building the app with build build optimizations on, we should not inline any sourcemaps at all.
>
> The inline sourcemaps are useful for dev mode and debugging runtime issue, but don't help prod debugging easier in a way that you couldn't use devmode for and only add a lot of weight.

File: packages/angular_devkit/build_angular/src/angular-cli-files/models/webpack-configs/styles.ts [hunk 1]
```diff
@@ -193,7 +193,13 @@ export function getStylesConfig(wco: WebpackConfigOptions) {
         options: {
           ident: 'embedded',
           plugins: postcssPluginCreator,
-          sourceMap: cssSourceMap && !buildOptions.sourceMap.hidden ? 'inline' : false,
+          sourceMap: cssSourceMap
+            // Never use component css sourcemap when optimizations are on.
+            // It will just increase bundle size without offering good debug experience.
+            && !buildOptions.optimization
```
Comment (alan-agius4) [inline packages/angular_devkit/build_angular/src/angular-cli-files/models/webpack-configs/styles.ts]: You should actually check it style optimization is enabled. As styles optimization can be disabled.

Comment (filipesilva) [reply packages/angular_devkit/build_angular/src/angular-cli-files/models/webpack-configs/styles.ts]: You're right, and that's also what was breaking a server builder test. Thanks!

**Comment by clydin** [comment]
Actually, any reason we can't create file source maps for the component stylesheets?

**Comment by filipesilva** [comment]
I'm not sure if the if the AOT compiler supports non-inline css sourcemaps actually. @IgorMinar @alxhub should we use them?

**Comment by clydin** [comment]
Would it matter at the AOT level?  It's just a different source map comment url.

**Comment by filipesilva** [comment]
I vaguely remember it mattered for AOT debugging, but not sure anymore. I think that's why we always inline them.

**Comment by alan-agius4** [comment]
I think the problem was that when not inlining it didn’t work for a number of use cases.

- Components styles sourcemaps didn’t work on SSR unless inlined.
- I don’t think that client bundles can have multiple sourcemaps urls.

**Comment by clydin** [comment]
The bundles wouldn't.  The injected style elements would each have one.  Instead of a data URL they would have a file URL.

**Pull request merged**